profiles.yml

```diff
@@ -0,0 +1,8 @@
+sparkutils:
+  target: test
+  outputs:
+    test:
+      type: spark
+      method: session
+      schema: test
+      host: NA        # not used, but required by `dbt-core`
```

Comment:
Ok, I see this requirement comes from `dbt-spark`, not `dbt-core`. Do you have an example of a field that is conditionally typed on the connection type? I could make new `Credentials` data class, but don't understand how it will be used. The `Plugin` [has a reference](https://github.com/dbt-labs/dbt-spark/blob/5ff1c42560fbf4265d9cc36eb18f18fff0f91877/dbt/adapters/spark/__init__.py#L11) to the `SparkCredentials only.

Reply:
And do you want to wait with making this field optional in `dbt-spark`? Or are you ok with merging this PR and cleaning this up later?

Reply:
No need for this to block the PR! Just wanted to record the clarification